hs_tools_resource/models.py

```diff
@@ -35,190 +33,112 @@ def metadata(self):
         md = ToolMetaData()
         return self._get_metadata(md)
 
-
 processor_for(ToolResource)(resource_processor)
 
 
 class RequestUrlBase(AbstractMetaDataElement):
-    term = 'Request Url Base'
-    value = models.CharField(null=True, max_length="500") # whatever the user gives us- format is "http://www.example.com/{resource-info}"
-
+    term = 'RequestUrlBase'
+    value = models.CharField(max_length=1024, null=True)
+    resShortID = models.CharField(max_length=128, default="UNKNOWN")
```

Comment:
@zhiyuli @shawncrawley I don't see that any of the above issues raised (most of them minor) have been addressed.  

The most problematic item is the (seeming) duplication of authorization logic (instead of using hs_core.views.utils.authorize()); this must be addressed before merging.  We cannot afford to have authorization logic proliferating.  If hs_core.views.utils.authorize() won't meet your needs, then we eight need to refactor hs_core.views.utils.authorize(), or create a new function that lives beside hs_core.views.utils.authorize().  I have to draw a line at the proliferation of authorization logic, as it can lead to bad unintentional outcomes (e.g. granting access when it should be), and it makes the system much harder to manage/re-factor going forward.

As for this line note, to map from webapp metadata instance to webapp resource instance, how is it that you have an webapp metadata instance that is orphaned from the webapp resource?  Couldn't you always discover the metadata from the resource, in which case, you'd always know the ID of the resource?  If not, @ironfroggy can you provide an example query to do the needed mapping?

Overall, I think part of the problem I am having is that I haven't seen a diagram of your data model, and also, none of your metadata classes have any comments describing what they are intended to do.  Sorry if this causes problems for including this in the AGU release, but as it stands, I don't feel that this code is understandable and maintainable in the long term (either by the original authors or by others).